From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: John Cupitt <jcupitt@gmail.com>
Date: Sun, 4 Sep 2022 16:35:00 +0100
Subject: [PATCH 1/1] check for blocked operations on cache lookup

We could return an operation from cache even after the operatuion had
been blocked. This was harmless, but could cause confusion.

see https://github.com/kleisauke/wasm-vips/pull/24

Upstream-Status: Accepted [https://github.com/libvips/libvips/commit/702ed8298f45d7ba342ebf5bae612d159e9cec6f]

diff --git a/libvips/iofuncs/cache.c b/libvips/iofuncs/cache.c
index 1111111..2222222 100644
--- a/libvips/iofuncs/cache.c
+++ b/libvips/iofuncs/cache.c
@@ -798,8 +798,10 @@ vips_cache_operation_lookup( VipsOperation *operation )
 	result = NULL;
 
 	if( (hit = g_hash_table_lookup( vips_cache_table, operation )) ) {
-		if( hit->invalid ) {
-			/* There but has been tagged for removal.
+		if( hit->invalid ||
+                        (VIPS_OPERATION_GET_CLASS( hit->operation )->flags &
+                                VIPS_OPERATION_BLOCKED) ) {
+			/* Has been tagged for removal, or has been blocked.
 			 */
 			vips_cache_remove( hit->operation );
 			hit = NULL;
